### PR TITLE
Add basic documentation

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    R.queueName = "notifications.user_signed_in",
+    R.body = "{\"user_id\": 1}"
+  }
+```
+
+#### Expiring messages
+Sends a `message` to the given `destination`. If nobody consumes the message in
+`timeoutInMs` milliseconds then the message is discarded. This is useful for showing
+notifications that must happen in a certain timeframe but where we don't really
+care if it reached the destination or not.
```
Comment (willmore) [inline README.md]: "but where we don't really care if it reached the destination or not" => "...but guaranteed delivery is not a strict requirement."

File: README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+```haskell
+  processMessage (Freddy.Delivery body replyWith failWith) = replyWith body
+
+  connection <- Freddy.connect "127.0.0.1" "/" "guest" "guest"
+  Freddy.respondTo connection "echo" processMessage
+```
+
+## Tapping into messages
+Listens for messages on a given destination or destinations without
+consuming them. It is useful for general messages that two or more clients
```
Comment (willmore) [inline README.md]: Would you want to use the term "broadcasting"?

**Comment by indrekj** [comment]
Updated

File: README.md (hunk 2 above)
Comment (indrekj) [inline README.md]: Removed that part instead. It is redundant anyway.

**Comment by indrekj** [comment]
!squash

**Pull request merged**